### PR TITLE
Sass 7308: Update model to match spec

### PR DESCRIPTION
**Pull request by ghousedev**
### Description
Update model to match spec

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/connectors/GetSavingsIncomeDataTysConnectorISpec.scala [hunk 1]
```diff
@@ -100,9 +100,10 @@ class GetSavingsIncomeDataTysConnectorISpec extends PlaySpec with WiremockSpec {
       "DES returns an empty 200" in {
         stubGetWithResponseBody(url, OK, returnedEmpty.toString())
         val result = await(connector.getSavingsIncomeData(nino, specificTaxYear))
-        val expectedResult = ErrorModel(INTERNAL_SERVER_ERROR, ErrorBodyModel.parsingError)
+        val expectedResult = Right(SavingsIncomeDataModel(None, None, None))
+          //ErrorModel(INTERNAL_SERVER_ERROR, ErrorBodyModel.parsingError)
```
Comment (ansahdavid) [inline it/connectors/GetSavingsIncomeDataTysConnectorISpec.scala:104]: Was this meant to be removed?

Comment (ghousedev) [reply it/connectors/GetSavingsIncomeDataTysConnectorISpec.scala:104]: Yes as it no longer returns a parsing error with all optional properties, returns the empty model like on line 103 instead.

Comment (ghousedev) [reply it/connectors/GetSavingsIncomeDataTysConnectorISpec.scala:104]: Yes it was meant to be removed my bad.

**Review by ansahdavid — DISMISSED** [review]

**Review by ansahdavid — APPROVED** [review]

File: app/controllers/CreateOrAmendSavingsController.scala [hunk 2]
```diff
@@ -35,12 +35,13 @@ class CreateOrAmendSavingsController @Inject()(createOrAmendSavingsService: Crea
 
   def createOrAmendSavings(nino: String, taxYear: Int): Action[AnyContent] = authorisedAction.async { implicit user =>
     user.request.body.asJson.map(_.validate[CreateOrAmendSavingsModel]) match {
-      case Some(JsSuccess(model, _)) => responseHandler(createOrAmendSavingsService.createOrAmendSavings(nino, taxYear, model))
```
Comment (basotia) [inline app/controllers/CreateOrAmendSavingsController.scala:38]: In this controller can we introduce correlationId change, as the 1903 spec does return 
`400 | INVALID_CORRELATION_ID` Also note the correlationId header name is different for MTD vs downstream, please implement it as per spec.

Comment (ghousedev) [reply app/controllers/CreateOrAmendSavingsController.scala:38]: Correlation Id should be passed from the frontend, or are we adding it here too?

**Review by basotia — APPROVED** [review]

**Pull request merged**